rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -109,8 +109,12 @@ NodeParameters::NodeParameters(
       rclcpp::ParameterMap initial_map = rclcpp::parameter_map_from(params);
 
       // Enforce wildcard matching precedence
-      // TODO(cottsay) implement further wildcard matching
-      const std::vector<std::string> node_matching_names{"/**", combined_name_};
+      const std::vector<std::string> node_matching_names{
+        "/**",
+        "/*/" + std::string(node_base->get_name()),
+        std::string(node_base->get_namespace()) + "/*",
```

Comment:
what if namespace is NOT specified, this will be `//*`. i think this is not handled as expected.